[Bug]:  panic on publish when no Github token is set

### What happened?

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14c0c17]

goroutine 10 [running]:
github.com/goreleaser/goreleaser/internal/client.(*githubClient).CreateFile(0xc000712070, 0xc0005b7830, {{0xc00043eab8, 0x0}, {0xc000043230, 0x0}}, {{0xc00043ea68, 0x6}, {0xc00043ea90, 0xa}}, ...)
        /Users/dirien/Tools/repos/goreleaser/internal/client/github.go:103 +0x277
github.com/goreleaser/goreleaser/internal/pipe/gofish.doPublish(0xc000210400, 0xc000822380, {0x27045f8, 0xc000712070})


### How can we reproduce this?

Dont set a GITHUB_TOKEN and call:

`goreleaser release --skip-announce --rm-dist --skip-validate`

### goreleaser version

```bash
0.180.2
```


### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

_No response_

--- a/internal/client/github.go
+++ b/internal/client/github.go
@@ -100,7 +100,7 @@ func (c *githubClient) CreateFile(
 		path,
 		&github.RepositoryContentGetOptions{},
 	)
-	if err != nil && res.StatusCode != 404 {
+	if err != nil && (res == nil || res.StatusCode != 404) {
 		return err
 	}
 

